DiffusionPreprocessing/DiffPreprocPipeline_Split.sh

```diff
@@ -0,0 +1,390 @@
+				Subject=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--posData=*)
+				PosInputImages=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--negData=*)
+				NegInputImages=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--posDataShimGroup=*)
+				PosShimLabels=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--negDataShimGroup=*)
+				NegShimLabels=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--dwiname=*)
+				DWIName=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			*)
+				PassOnArguments+=" ${argument} "
+				index=$(( index + 1 ))
+				;;
+		esac
+	done
+	
+	local error_msgs=""
+
+	# check required parameters
+	if [ -z ${PosInputImages} ] ; then
+		error_msgs+="\nERROR: <positive-phase-encoded-data> not specified"
+	fi
+	
+	if [ -z ${NegInputImages} ] ; then
+		error_msgs+="\nERROR: <negative-phase-encoded-data> not specified"
+	fi
+
+	if [ -z ${PosShimLabels} ] ; then
+		error_msgs+="\nERROR: <shim-group-for-positive-phase-encoding-data> not specified"
+	fi
+	
+	if [ -z ${NegShimLabels} ] ; then
+		error_msgs+="\nERROR: <shim-group-for-negative-phase-encoding-data> not specified"
+	fi
+
+	if [ -z ${StudyFolder} ] ; then
+		error_msgs+="\nERROR: <study-path> not specified"
+	fi
+
+	if [ -z ${Subject} ] ; then
+		error_msgs+="\nERROR: <subject-id> not specified"
+	fi
+
+	if [ ! -z "${error_msgs}" ] ; then
+		usage
+		echo -e ${error_msgs}
+		echo ""
+		exit 1
+	fi
+	
+	# report parameters
+	echo "-- ${SCRIPT_NAME}: Specified Command-Line Parameters - Start --"
+	echo "   PosInputImages: ${PosInputImages}"
+	echo "   NegInputImages: ${NegInputImages}"
+	echo "   DWIName: ${DWIName}"
+	echo "   Arguments for DiffPreprocPipeline.sh: ${PassOnArguments}"
+	echo "-- ${SCRIPT_NAME}: Specified Command-Line Parameters - End --"
+}
+
+# 
+# Function Description
+#  Validate necessary environment variables
+#
+validate_environment_vars()
+{
+	local error_msgs=""
+
+	# validate
+	if [ -z ${HCPPIPEDIR_dMRI} ] ; then
+		error_msgs+="\nERROR: HCPPIPEDIR_dMRI environment variable not set"
+	fi
+
+	if [ ! -e ${HCPPIPEDIR}/DiffusionPreprocessing/DiffPreprocPipeline.sh ] ; then
+		error_msgs+="\nERROR: HCPPIPEDIR/DiffusionPreprocessing/DiffPreprocPipeline.sh not found"
+	fi
+
+	if [ -z ${FSLDIR} ] ; then
+		error_msgs+="\nERROR: FSLDIR environment variable not set"
+	fi
+	
+	if [ ! -z "${error_msgs}" ] ; then
+		usage
+		echo -e ${error_msgs}
+		echo ""
+		exit 1
+	fi
+
+	# report
+	echo "-- ${SCRIPT_NAME}: Environment Variables Used - Start --"
+	echo "   HCPPIPEDIR_dMRI: ${HCPPIPEDIR_dMRI}"
+	echo "   FSLDIR: ${FSLDIR}"
+	echo "-- ${SCRIPT_NAME}: Environment Variables Used - End --"
+}
+
+#
+# Function description
+#  Main processing of script if --split is set
+#
+main()
+{
+	# Get Command Line Options
+	get_options $@
+
+	# Validate environment variables
+	validate_environment_vars $@
+
+	# Establish tool name for logging
+	log_SetToolName "${SCRIPT_NAME}"
+
+
+    input_dwinames=""
+    while read -r line ; do
```

Comment:
Comment preceding this section would be helpful.
i.e., that "while read -r line" is being feed the output "split_shimgroups.py" as input lines.